src/shapes/text.class.js

```diff
@@ -1574,7 +1576,19 @@
    * @param {Function} [callback] Callback to invoke when an fabric.Text instance is created
    */
   fabric.Text.fromObject = function(object, callback) {
-    return fabric.Object._fromObject('Text', object, callback, 'text');
+    var path = object.path;
+    delete object.path;
```

Comment:
i made a wrong suggestion here.
In this case we are mutating the input of the function, meaning that the second time you restore the same object, we deleted the path property from it.
We need to clone this object before deleting the path. ( fabric.util.object.clone is what we use generally )